Week8/app/src/main/java/com/isabelledionisius/myomelettapp/omeletts/MainActivity.kt

```diff
@@ -0,0 +1,108 @@
+package com.isabelledionisius.myomelettapp.omeletts
+
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.view.Gravity
+import android.widget.Toast
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import androidx.work.*
+import com.isabelledionisius.myomelettapp.R
+import com.isabelledionisius.myomelettapp.networking.RemoteApi
+import com.isabelledionisius.myomelettapp.omelettdetails.DetailActivity
+import com.isabelledionisius.myomelettapp.omeletts.list.OmelettAdapter
+import com.isabelledionisius.myomelettapp.omeletts.model.Omelett
+import com.isabelledionisius.myomelettapp.worker.PeriodicWorker
+import androidx.work.PeriodicWorkRequestBuilder
+import kotlinx.android.synthetic.main.activity_main.*
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.launch
+import java.util.concurrent.TimeUnit
+
+
+/** Main activity of the list page
+ *
+ */
+class MainActivity : AppCompatActivity(),
+    OmelettAdapter.OmelettListClickListener {
+
+ private lateinit var omelettListRecyclerView: RecyclerView
+
+    companion object {
+        const val INTENT_LIST_KEY = "list"
+    }
+
+    /**
+     * Creating the adapter and assigning the values to it
+     */
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+        omelettListRecyclerView = this.recycler_view//findViewById(R.id.recycler_view)
+        omelettListRecyclerView.layoutManager = LinearLayoutManager(this)
+        getData()
+        synchronizeData()
+    }
+
+    private fun getData() {
+        GlobalScope.launch {
```

Comment:
Not like this. We have LifecycleScope for this. GlobalScope should be avoided whenever possible.